router.go

```diff
@@ -296,7 +305,14 @@ func (r Router) findBestRoute(path string) (bestRoute *route, tokens []string) {
 			}
 		}
 	}
-	return bestRoute, tokens
+	if len(parts) > 1 {
```

Comment:
This is a nitpick, but I would feel more comfortable if we parsed the query parameters outside of the `findBestRoute` function. They don't actually have any affect on choosing the best route.